activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -1221,7 +1221,7 @@ def configure_connection
 
         def attempt_configure_connection
           configure_connection
-        rescue
+        rescue Exception # Need to handle things such as Timeout::ExitException
```

Comment:
Should we use `ensure`, to cover the version range where Timeout used `throw`?

Reply:
Hum, you have a point. I haven't tested with older versions of timeout.

Reply:
Actually since https://github.com/rails/rails/commit/5fbaa524b99 (7.1) we do require `timeout >= 0.4`.

So I guess we're fine. It's a bit weird we depend on timeout even though we don't need it though.